src/sentry/utils/snuba.py

```diff
@@ -984,11 +983,6 @@ def constrain_condition_to_dataset(cond, dataset):
             if name is None:
                 return None
             cond[0] = name
-            # Reformat 32 byte uuids to 36 byte variants.
-            # The transactions dataset requires properly formatted uuid values.
-            # But the rest of sentry isn't aware of that requirement.
-            if dataset == Dataset.Transactions and name == "event_id" and len(cond[2]) == 32:
-                cond[2] = six.text_type(uuid.UUID(cond[2]))
```

Comment:
If you commit this queries will fail, and you cannot assume Snuba and Sentry will be deployed at the same time so the change has to be backward compatible on both sides, you can add an intermediate virtual column (like we do for issue/group_id) to ensure both sentry and snuba work during the process.

Second problem is that we need to be sure we are not compromising the index performance by adding the replaceAll function. We should test that WHERE event_id = XXX is the same as WHERE replaceAll(event_id ,...) = XXX same for order by. In some cases this could turn an index search into a full table scan.

Reply:
I don't think the cost of adding backwards compatibility is strictly required here. With the amount of usage that discover2 is getting, I'm ok with it being a little broken for a deploy window sized amount of time.

Reply:
So the second issue is not a problem as of now since event_id is not in the order_by clause nor in any index for transactions.

If you are ok in avoiding the work to ensure backward compatibility, then this is fine to me.